parquet/src/main/java/org/apache/iceberg/parquet/MessageTypeToType.java

```diff
@@ -96,11 +96,13 @@ public Type struct(GroupType struct, List<Type> fieldTypes) {
 
   @Override
   public Type list(GroupType array, Type elementType) {
-    GroupType repeated = array.getType(0).asGroupType();
-    org.apache.parquet.schema.Type element = repeated.getType(0);
+    org.apache.parquet.schema.Type repeated = array.getType(0);
+    org.apache.parquet.schema.Type repeatedElement = array.getFields().get(0);
+    boolean isElementType = ParquetSchemaUtil.isListElementType(repeatedElement, array.getName());
```

Comment:
Is elementType here really is "isBackwardCompatibleListElement" or something like that? I think we should rename this to make it more clear.

Reply:
To be consistent with `IsOldListElementType` method name, I will rename this to `isOldListElementType`